### PR TITLE
Create an endpoint to get confirmed goods delivery

### PR DESCRIPTION
**Pull request by marynakernychna**

### CODE REVIEW

**Review by Eugene-pas — APPROVED** [review]

File: Core/Specifications/OfferSpecification.cs [hunk 1]
```diff
@@ -150,7 +152,22 @@ public GetByIdWithTrip(int offerId, string userId)
                     .Include(o => o.Trip);
             }
         }
-        
+
+        internal class GetConfirmedByTripAndUserIds : Specification<Offer>
```
Comment (Grasmus) [inline Core/Specifications/OfferSpecification.cs]: It does not pull confirmed offers

**Review by AndriiChepeliuk — APPROVED** [review]

File: API/Controllers/OfferController.cs [hunk 2]
```diff
@@ -62,6 +63,18 @@ public async Task<IActionResult> GetNearRouteAsync([FromQuery] TripIdDTO tripIdD
             return Ok(await _offerService.GetNearRouteAsync(tripIdDTO.TripId));
         }
 
+        [HttpGet("confirmed-goods-delivery")]
```
Comment (Grasmus) [inline API/Controllers/OfferController.cs]: Change route name. How can we know that it refers to driver?

File: API/Controllers/OfferController.cs [hunk 3]
```diff
@@ -62,6 +63,18 @@ public async Task<IActionResult> GetNearRouteAsync([FromQuery] TripIdDTO tripIdD
...
+        [AuthorizeByRole(IdentityRoleNames.User)]
+        public async Task<IActionResult> GetConfirmedGoodsDeliveryAsync(
```
Comment (Grasmus) [inline API/Controllers/OfferController.cs]: Does we get only confirmed offers??

File: API/Controllers/OfferController.cs [hunk 4]
```diff
@@ -62,6 +63,18 @@ public async Task<IActionResult> GetNearRouteAsync([FromQuery] TripIdDTO tripIdD
...
+            [FromQuery] TripIdDTO tripIdDTO)
+        {
+            var userId = _userService.GetCurrentUserNameIdentifier(User);
+            var offers =
+                await _offerService.GetConfirmedGoodsDeliveryAsync(tripIdDTO.TripId, userId);
```
Comment (Grasmus) [inline API/Controllers/OfferController.cs]: Same as controller

**Review by Grasmus — APPROVED** [review]

**Pull request merged**